### PR TITLE
Add Logout Reason to Revoke API call.

### PR DESCRIPTION
**Pull request by brandonpage**
I have elected not to add support for passing the logout reason from hybrid or React Native apps because the audience for this change is internal apps and none of them use those libraries anymore.

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java [hunk 1]
```diff
@@ -606,7 +606,8 @@ public String getNewAuthToken() {
                         	if (Looper.myLooper() == null) {
                                 Looper.prepare();
                         	}
-                            SalesforceSDKManager.getInstance().logout(null, false);
+                            SalesforceSDKManager.getInstance()
+                                    .logout(null, null, false, OAuth2.LogoutReason.TOKEN_EXPIRED);
```
Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java:610]: When the refresh token has expired, there won't be a (successful) client call to the server to revoke the token and therefore the logout reason will never be sent to the server. But the server should have created a log line when the refresh token expires.

TLDR: TOKEN_EXPIRED reason can never be logged by the client only by the server

Comment (wmathurin) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java:610]: That being said, it's fine to use that reason here (even though it won't leave the client).

Comment (brandonpage) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java:610]: Great callout.  I think I will leave it as there are no negative side affects but I will add a comment noting this.

Comment (bbirman) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java:610]: I'll update on iOS too

**Review by wmathurin — APPROVED** [review]

**Comment by brandonpage** [comment]
I have fixed he (unrelated) broken `SalesforceSDK` tests in #2575.
